worker.go

```diff
@@ -31,8 +32,14 @@ import (
 type (
 	// Worker represents objects that can be started and stopped.
 	Worker interface {
-		Stop()
+		// Start starts the worker in a non-blocking fashion
 		Start() error
+		// Run is a blocking start and cleans up resources when killed
+		Run()
```

Comment:
I think we should merge Run() and Start(). Offering too much feature confuse people. For those who don't want to be blocked, can easily do that by using a separate goroutinue to call Start().